crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1143,7 +1145,39 @@ def upgrade_all_compute_nodes(upgrade_mode = :non_disruptive)
         end
       end
 
-      def upgrade_compute_nodes(virt)
+      def parallel_upgrade_compute_nodes(compute_nodes)
+        save_upgrade_state("Entering disruptive upgrade of compute nodes")
+        save_nodes_state(compute_nodes, "compute", "upgrading")
+        begin
+          execute_scripts_and_wait_for_finish(
+            compute_nodes,
+            "/usr/sbin/crowbar-upgrade-os.sh",
+            900
+          )
+          save_upgrade_state("Repositories prepared successfully.")
+        rescue StandardError => e
+          raise_node_upgrade_error(
+            "Error while upgrading compute nodes. " + e.message
+          )
+        end
+        # FIXME: should we paralelize this as well?
```

Comment:
Yeah, especially having the ```join_and_chef``` part  running in parallel could speed things up quite a bit. But it's ok to address that later IMO, just add it to the backlog.